cluster/watchdog.go

```diff
@@ -47,7 +47,9 @@ func (w *Watchdog) removeDuplicateContainers(e *Engine) {
 			if containerInCluster.Config.SwarmID() == container.Config.SwarmID() && containerInCluster.Engine.ID != container.Engine.ID {
 				log.Debugf("container %s was rescheduled on node %s, removing it", container.Id, containerInCluster.Engine.ID)
 				// container already exists in the cluster, destroy it
-				e.RemoveContainer(container, true, true)
+				if err := e.RemoveContainer(container, true, true); err != nil {
+					log.Errorf("Failed to remove duplicate container %s on node %s: %v", container.Id, containerInCluster.Engine.ID, err)
```

Comment:
I think printing engine name improves readability. It's possible multiple engine has same name, but that's bad practice.